pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -327,7 +327,7 @@ func isNoOp(s state) bool {
 	return ok
 }
 
-func threadID(didCommMsg service.DIDCommMsg) (string, error) {
+func threadID(didCommMsg *service.DIDCommMsg) (string, error) {
```

Comment:
I suggest make threadID function part of service.DIDCommMsg.

Reply:
I have created PR for that https://github.com/hyperledger/aries-framework-go/pull/465